### PR TITLE
added new functionality to read sim_telarray calibration events

### PR DESCRIPTION
**Pull request by GernotMaier**
New functionality to read sim_telarray calibration events.

Example:

```
with open_hessio('gamma_20deg_180deg_run4___cta-prod3-demo_desert-2150m-Paranal-baseline.simtel') as event:
   for event_id in event.move_to_next_calib_event(limit=5):
        print('run id {}:, event number: {}'.format(event.get_run_number() , event_id))
        print('    Triggered telescopes for this event: {}'.format(event.get_telescope_with_data_list()))
        print('    balkds {}'.format(event.get_adc_sample(5)))
```

This allows to read out the files with NSB events only.

### CODE REVIEW

**Comment by kosack** [comment]
I think rather than adding another iteration function and making pyhessio more complex, it  would be much better to make this more general, e.g. to have `move_to_next_event()` simply have a option to tell it which event types to consider (and have the logic later in ctapipe).  This implementation means we have to invent multiple new event sources in ctapipe, rather than reuse the one we have with a simple new option.  Here we would have to choose "normal event" *or* "calib events", however there is the case in real data that we have both interleaved, so you'd have to add yet another iterator like "move_to_next_normal_or_calib_event()"...

How about just the following:
```py3
with open_hessio(filename) as events:
    events.set_allowed_types([pyhessio.event_type.cherenkov, pyhessio.event_type.calib])
    for event_id in event.move_to_next_event(limit=5):
         if event.get_type() == pyhessio.event_type.calib:
              # do something

```

The default would be to read only `pyhessio.event_type.cherenkov` ty[e evemts

**Comment by kosack** [comment]
For the types, I think just exposing the event type as an integer is good to start, and then we can always define an enum to make them more readable, something like:
```py3
from enum import Enum

class EventType(Enum):
    CHERENKOV = 0
    PEDESTAL = 1
    FLATFIELD = 2
    LASER = 3
```
or something like that, then you can use them in python by:

```py3
if pyhessio.get_event_type() == EventType.CHERENKOV.value:
   pass
```

**Comment by GernotMaier** [comment]
I've introduced now EventTypes as enums and assigned them as integer values the corresponding values of the hessio event types (2010, 2021, 2023, 2028, as defined in hessioxxx/include/io_hess.h)

There is now one single function for all event types:

```
with open_hessio(filename) as events:
    for event_id in event.move_to_next_event(limit=5, event_type = EventType.CHERENKOV.value ):
              # do something
```

Any code calling move_to_next_mc_event and  fill_next_mc_event should be changed to

- fill_next_event( event_type = EventType.MC.value)
- event.move_to_next_event(limit=0, event_type = EventType.CHERENKOV.value )

Note that this solution does not allow (yet) to read out e.g. CHERENKOV and PEDESTAL events.

File: pyhessio/tests/test_hessio.py [hunk 1]
```diff
@@ -313,7 +312,7 @@ def test_hessio():
         hessio.open_file("pyhessio-extra/datasets/gamma_test.simtel.gz")
         # Testing move_to_next_mc_event iterator
         #event_id = next(hessio.move_to_next_mc_event())
-        event_id = hessio.fill_next_mc_event()
+        event_id = hessio.fill_next_event( EventType.MC.value )
         run_number = hessio.get_run_number()
 
         assert run_number == 31964
```
Comment (jacquemier) [inline pyhessio/tests/test_hessio.py]: Couyld you please add unit test with calibration event ?

**Comment by GernotMaier** [comment]
Calibration events are needed for the unit tests:

- please acknowledge pull request https://github.com/cta-observatory/pyhessio-extra/pull/1
- I will then upload some units tests for calibration events

**Pull request merged**